### PR TITLE
quick fix for kubernetes above 1.16

### PR DESCRIPTION
**Pull request by t3476**

### CODE REVIEW

**Comment by ShadowJonathan** [comment]
Would this effectively lock `caddyserver/ingress` to kubernetes 1.16 and up?

**Comment by hairyhenderson** [comment]
@ShadowJonathan no, `Deployment`s in `apps/v1` has been available since Kubernetes v1.9.

See https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/ for details

File: kubernetes/helm/caddyingresscontroller/templates/clusterrole.yaml [hunk 1]
```diff
@@ -20,9 +20,9 @@ rules:
       - pods
       - nodes
       - routes
-      - extensions
```
Comment (hairyhenderson) [inline kubernetes/helm/caddyingresscontroller/templates/clusterrole.yaml:23]: we'll still need `extensions/v1beta1` to see `Ingress` for pre-v1.14 versions of Kubernetes - is there any reason to _not_ support pre-v1.14?

**Comment by Embraser01** [comment]
PR #31 rewrite the entire Helm chart and should be ready soon (cc @mavimo). It would be nice if you could test it and give us feedback on it 🙂

@hairyhenderson regarding 
> we'll still need extensions/v1beta1 to see Ingress for pre-v1.14 versions of Kubernetes - is there any reason to not support pre-v1.14?

I'd say we don't support it as it'll be deprecated on [AWS](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html) by June and other providers have support for v1.14+

Moreover, we now use the Ingress resources from `networking.k8s.io/v1beta1` available in v1.14+ clusters

**Comment by hairyhenderson** [comment]
@Embraser01 ah, ok. That's fair. I'm mostly using v1.15 in production, though we have an on-prem product that needs to support v1.11+ due to OpenShift support.

**Comment by ShadowJonathan** [comment]
Maybe that new requirement should be noted down somewhere in the readme file then? Preferably with a commit in this PR

File: kubernetes/helm/caddyingresscontroller/templates/clusterrole.yaml (hunk 1 above)
Comment (t3476) [reply kubernetes/helm/caddyingresscontroller/templates/clusterrole.yaml:23]: IBM and Google also deprecated v1.16

**Comment by mavimo** [comment]
@t3476 @ShadowJonathan @hairyhenderson the current version of the chart work with kubernetes 1.18 (see https://github.com/caddyserver/ingress/runs/792277748?check_suite_focus=true ) and is tested in CI, I'm going to close this PR, let me know if you think something is missing, thx!

**Pull request closed without merging**